docs/userguide/storagedriver/aufs-driver.md

```diff
@@ -0,0 +1,194 @@
+
+AUFS stacks multiple directories and exposes them as a unified view through a single mount point. All of the directories in the stack, as well as the union mount point, must all exist on the same Linux host. AUFS refers to each directory that it stacks as a *branch*.
+
+Within Docker, AUFS union mounts enable image layering. The AUFS storage driver implements Docker image layers using this union mount system. AUFS branches correspond to Docker image layers. The diagram below shows a Docker container based on the `ubuntu:latest` image.
+
+![](images/aufs_layers.jpg)
+
+This diagram shows the relationship between the Docker image layers and the AUFS branches (directories) in `/var/lib/docker/aufs`. Each image layer  and the container layer correspond to an AUFS branch (directory) in the Docker host's local storage area. The union mount point gives the unified view of all layers.
+
+AUFS also supports the copy-on-write technology (CoW). Not all storage drivers do.
+
+## Container reads and writes with AUFS
+
+Docker leverages AUFS CoW technology to enable image sharing and minimize the use of disk space. AUFS works at the file level. This means that all AUFS CoW operations copy entire files - even if only a small part of the file is being modified. This behavior can have a noticeable impact on container performance, especially if the files being copied are large, below a lot of image layers, or the CoW operation must search a deep directory tree.
+
+Consider, for example, an application running in a container needs to add a single new value to large key-value store (file). If this is the first time the file is modified it does not yet exist in the container's top writable layer. So, the CoW must *copy up* the file from the underlying image. The AUFS storage driver searches each image layer for the file. The search order is from top to bottom. When it is found, the entire file is *copied up* to the container's top writable layer. From there, it can be opened and modified.
+
+Larger files obviously take longer to *copy up* than smaller files, and files that exist in lower image layers take longer to locate than those in higher layers. However, a *copy up* operation only occurs once per file on any given container. Subsequent reads and writes happen against the file's copy already *copied-up* to the container's top layer.
+
+
+## Deleting files with the AUFS storage driver
+
+The AUFS storage driver deletes a file from a container by placing a *whiteout
+file* in the container's top layer. The whiteout file effectively obscures the
+existence of the file in image's lower, read-only layers.  The simplified
+diagram below shows a container based on an image with three image layers.
+
+![](images/aufs_delete.jpg)
+
+The `file3` was deleted from the container. So, the AUFS storage driver  placed
+a whiteout file in the container's top layer. This whiteout file effectively
+"deletes"  `file3` from the container by obscuring any the original file's
```

Comment:
by obscuring any the -> by obscuring any of the

Also, sentence looks incomplete